app/dao/fact_billing_dao.py

```diff
@@ -19,11 +24,155 @@
+    )
+    return query
+
+
+def fetch_sms_billing_for_all_services(start_date, end_date):
+
+    # ASSUMPTION: AnnualBilling has been populated for year.
+    billing_year = which_financial_year(start_date)
+    free_allowance_remainder = fetch_sms_free_allowance_remainder(start_date).subquery()
+    sms_billable_units = func.sum(FactBilling.billable_units * FactBilling.rate_multiplier)
+    sms_remainder = func.coalesce(free_allowance_remainder.c.sms_remainder, AnnualBilling.free_sms_fragment_limit)
+    chargeable_sms = case([(sms_remainder == 0, sms_billable_units),
+                           (sms_billable_units - sms_remainder <= 0, 0),
+                           (sms_billable_units - sms_remainder > 0,
+                            sms_billable_units - sms_remainder)], else_=0)
+    sms_cost = case([(sms_remainder == 0, sms_billable_units * FactBilling.rate),
+                     (sms_billable_units - sms_remainder <= 0, 0),
+                     (sms_billable_units - sms_remainder > 0, (sms_billable_units - sms_remainder) * FactBilling.rate)
+                     ], else_=0)
+
+    query = db.session.query(
+        Organisation.name.label('organisation_name'),
+        Organisation.id.label('organisation_id'),
+        Service.name.label("service_name"),
+        FactBilling.service_id.label("service_id"),
+        AnnualBilling.free_sms_fragment_limit,
+        FactBilling.rate.label('sms_rate'),
+        sms_remainder.label("sms_remainder"),
+        sms_billable_units.label('sms_billable_units'),
+        chargeable_sms.label("chargeable_billable_sms"),
+        sms_cost.label('sms_cost'),
+    ).join(
+        Service.annual_billing,
+    ).outerjoin(
+        free_allowance_remainder, Service.id == free_allowance_remainder.c.service_id
+    ).outerjoin(
+        Organisation, Service.organisation_id == Organisation.id
+    ).filter(
+        FactBilling.service_id == Service.id,
+        FactBilling.bst_date >= start_date,
+        FactBilling.bst_date <= end_date,
+        FactBilling.notification_type == SMS_TYPE,
+        AnnualBilling.financial_year_start == billing_year,
+    ).group_by(
+        Organisation.name,
+        Organisation.id,
+        FactBilling.service_id,
+        Service.name,
+        AnnualBilling.free_sms_fragment_limit,
+        free_allowance_remainder.c.sms_remainder,
+        FactBilling.rate,
+    ).order_by(
+        Organisation.name,
+        Service.name
+    )
+
+    return query.all()
+
+
+def fetch_letter_costs_for_all_services(start_date, end_date):
+    query = db.session.query(
+        Organisation.name.label("organisation_name"),
+        Organisation.id.label("organisation_id"),
+        Service.name.label("service_name"),
+        FactBilling.service_id.label("service_id"),
+        func.sum(FactBilling.notifications_sent * FactBilling.rate).label("letter_cost")
+    ).outerjoin(
+        Organisation, Service.organisation_id == Organisation.id
```

Comment:
similar join comment. This table uses Organisation, Service and FactBilling so those should be joined.